nilearn/datasets/neurovault.py

```diff
@@ -1507,11 +1513,42 @@ def _download_image_nii_file(image_info, collection, download_params):
         collection['relative_path'], image_file_name)
     image_absolute_path = os.path.join(
         collection['absolute_path'], image_file_name)
-    _simple_download(
-        image_url, image_absolute_path,
-        download_params['temp_dir'], verbose=download_params['verbose'])
+
+    resampled_image_file_name = 'image_{0}_resampled.nii.gz'.format(image_id)
+    resampled_image_absolute_path = os.path.join(
+        collection['absolute_path'], resampled_image_file_name)
+    resampled_image_relative_path = os.path.join(
+        collection['relative_path'], resampled_image_file_name)
+
     image_info['absolute_path'] = image_absolute_path
     image_info['relative_path'] = image_relative_path
+    image_info['resampled_absolute_path'] = resampled_image_absolute_path
+    image_info['resampled_relative_path'] = resampled_image_relative_path
+
+    if download_params['resample']:
+        # Generate a temporary file name
+        struuid = str(uuid.uuid1())
+
+        tmp_file = 'tmp_{0}.nii.gz'.format(struuid)
+
+        tmp_path = os.path.join(
+            collection['absolute_path'], tmp_file)
+
+        _simple_download(
+            image_url, tmp_path,
+            download_params['temp_dir'], verbose=download_params['verbose'])
+
+        # Resample here
+        print('Resampling...')
+        im_resampled = resample_img(img=tmp_path, target_affine=STD_AFFINE)
+        im_resampled.to_filename(resampled_image_absolute_path)
+
+        # Remove temporary file
+        os.remove(tmp_path)
```

Comment:
could we maybe put the above in a try/finally block? to make sure the temp file is always removed

Reply:
or if you prefer you can place the tempfile in `download_params['temp_dir']`. also, not sure you need the uuid, just appending something like "_not_resampled.nii.gz" should be fine. but using a uuid is also fine.